### ISSUE TITLE
Darwin: `cmake-boot` fails to build on `staging-next`

### ISSUE THREAD

**Issue by FRidh** [NixOS/nixpkgs#56244]
## Issue description

```
Scanning dependencies of target OSXScriptLauncher
[ 83%] Building CXX object Source/CMakeFiles/OSXScriptLauncher.dir/CPack/OSXScriptLauncher.cxx.o
[ 83%] Linking CXX executable ../bin/OSXScriptLauncher
[ 83%] Built target OSXScriptLauncher
Scanning dependencies of target cmakexbuild
[ 83%] Building CXX object Source/CMakeFiles/cmakexbuild.dir/cmakexbuild.cxx.o
[ 83%] Linking CXX executable ../bin/cmakexbuild
ld: framework not found CoreServices
clang-4.0: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [Source/CMakeFiles/cmakexbuild.dir/build.make:96: bin/cmakexbuild] Error 1
make[1]: *** [CMakeFiles/Makefile2:2160: Source/CMakeFiles/cmakexbuild.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
builder for '/nix/store/62agbb042zndr5xy2iszq7mlhfwbr9g3-cmake-boot-3.13.4.drv' failed with exit code 2
```

https://hydra.nixos.org/build/89378303

Relevant `staging-next` PR https://github.com/NixOS/nixpkgs/pull/56200

cc @NixOS/darwin-maintainers 

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

**Comment by LnL7** [comment]
I'll try to take a look today, I think we've had a very similar issue before tho.

**Comment by LnL7** [comment]
Apparently I fixed it before 02dcc8d734302b22ba635844c1c2a57ddca83853, I'll do some testing.

**Comment by matthewbauer** [comment]
Cmake is probably bundling libuv now… there might be a flag to not do that
